grype/db/curator.go

```diff
@@ -312,3 +328,49 @@ func (c *Curator) activate(dbDirPath string) error {
 	// activate the new db cache
 	return file.CopyDir(c.fs, dbDirPath, c.dbDir)
 }
+
+// ListingFromURL loads a Listing from a URL.
+func (c Curator) ListingFromURL() (curation.Listing, error) {
+	tempFile, err := afero.TempFile(c.fs, "", "grype-db-listing")
+	if err != nil {
+		return curation.Listing{}, fmt.Errorf("unable to create listing temp file: %w", err)
+	}
+	defer func() {
+		err := c.fs.RemoveAll(tempFile.Name())
+		if err != nil {
+			log.Errorf("failed to remove file (%s): %w", tempFile.Name(), err)
+		}
+	}()
+
+	// download the listing file
+	err = c.downloader.GetFile(tempFile.Name(), c.listingURL)
+	if err != nil {
+		return curation.Listing{}, fmt.Errorf("unable to download listing: %w", err)
+	}
+
+	// parse the listing file
+	listing, err := curation.NewListingFromFile(c.fs, tempFile.Name())
+	if err != nil {
+		return curation.Listing{}, err
+	}
+	return listing, nil
+}
+
+func defaultHTTPClient(fs afero.Fs, caCertPath string) (*http.Client, error) {
+	httpClient := cleanhttp.DefaultClient()
```

Comment:
getter test should use this as well

Reply:
unfortunately the getter tests don't create their own HTTP Client, they are retrieved from the httptest server object (`server.Client()`). So there isn't a nice/easy way to use this default client in testing.